applications/DEMApplication/custom_processes/apply_forces_and_moments_process.hpp

```diff
@@ -0,0 +1,239 @@
+
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// Execute method is used to execute the ApplyForcesAndMomentsProcess algorithms.
+    void Execute() override
+    {
+    }
+
+//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    void ExecuteInitializeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+
+        const double time = mrModelPart.GetProcessInfo()[TIME];
+
+        if(!mInterval.IsInInterval(time)) return;
+
+        const int nnodes = static_cast<int>(mrModelPart.Nodes().size());
+
+        if(nnodes != 0)
+        {
+            ModelPart::NodesContainerType::iterator it_begin = mrModelPart.NodesBegin();
+
+            #pragma omp parallel for
+            for(int i = 0; i<nnodes; i++) {
+                ModelPart::NodesContainerType::iterator it_node = it_begin + i;
+
+                array_1d<double, 3>& force = it_node->FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE);
+                array_1d<double, 3>& moment = it_node->FastGetSolutionStepValue(EXTERNAL_APPLIED_MOMENT);
+
+                for(int i=0; i<3; i++) {
+                    double force_value = 0.0;
+                    if(mForceValueIsNumeric[i]) {
+                        force_value = mForceValues[i];
+                    }
+                    else {
+                        force_value = mForceFunctions[i].CallFunction(it_node->X(), it_node->Y(), it_node->Z(), time);
+                    }
+                    force[i] = force_value;
+
+                    double moment_value = 0.0;
+                    if(mMomentValueIsNumeric[i]) {
+                        moment_value = mMomentValues[i];
+                    }
+                    else {
+                            moment_value = mMomentFunctions[i].CallFunction(it_node->X(), it_node->Y(), it_node->Z(), time);
+                    }
+                    moment[i] = moment_value;
+                }
+            }
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    void ExecuteFinalizeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+
+        const double time = mrModelPart.GetProcessInfo()[TIME];
+
+        if(mInterval.IsInInterval(time)) return;
+
+        const int nnodes = static_cast<int>(mrModelPart.Nodes().size());
+
+        if(nnodes != 0)
+        {
+            ModelPart::NodesContainerType::iterator it_begin = mrModelPart.NodesBegin();
+
+            #pragma omp parallel for
+            for(int i = 0; i<nnodes; i++) {
+                ModelPart::NodesContainerType::iterator it_node = it_begin + i;
+                it_node->FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE) = ZeroVector(3);
+                it_node->FastGetSolutionStepValue(EXTERNAL_APPLIED_MOMENT) = ZeroVector(3);
+            }

```

Comment:
please use the parallel utilites

Reply:
FYI I updated the corresponding Wiki today: https://github.com/KratosMultiphysics/Kratos/wiki/Parallel-Utilities